syntax = "proto3";
package parsers;

message User {
    enum Gender {
        MALE = 0;
        FEMALE = 1;
        OTHER = 2;
    }
    uint64 user_id = 1;
    string username = 2;
    uint32 birthday = 3;
    Gender gender = 4;
}

message Snapshot {
    uint64 uuid = 1;
    uint64 datetime = 2;
    User user = 3;
    Pose pose = 4;
    ColorImage color_image = 5;
    DepthImage depth_image = 6;
    Feelings feelings = 7;
}

message Pose {
    message Translation {
        double x = 1;
        double y = 2;
        double z = 3;
    }
    message Rotation {
        double x = 1;
        double y = 2;
        double z = 3;
        double w = 4;
    }
    Translation translation = 1;
    Rotation rotation = 2;
}

message ColorImage {
    uint32 width = 1;
    uint32 height = 2;
    string path = 3;
}

message DepthImage {
    uint32 width = 1;
    uint32 height = 2;
    string path = 3;
}

message Feelings {
    float hunger = 1;
    float thirst = 2;
    float exhaustion = 3;
    float happiness = 4;
}
